Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -772,6 +772,8 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseStatementList
         FALLTHROUGH;
     case AWAIT:
     case YIELD: {
+        semanticFailIfTrue(match(YIELD) && m_parserState.isParsingClassStaticBlock, "Cannot use 'yield' within class static block");
+        semanticFailIfTrue(match(AWAIT) && !findClosestAsyncFunctionScopeUntilStaticBlock(), "Cannot use 'await' within static block scope");
```

Comment:
This `findClosestAsyncFunctionScopeUntilStaticBlock` is too costly, see comments.